modules/cli/src/main/scala/scala/cli/commands/repl/Repl.scala

```diff
@@ -268,24 +279,28 @@ object Repl extends ScalaCommand[ReplOptions] {
   private def runRepl(
     options: BuildOptions,
     programArgs: Seq[String],
-    artifacts: Artifacts,
-    mainJarOrClassDir: Option[os.Path],
+    allArtifacts: Seq[Artifacts],
+    mainJarsOrClassDirs: Seq[os.Path],
     directories: scala.build.Directories,
     logger: Logger,
     allowExit: Boolean,
     dryRun: Boolean,
     runMode: RunMode.HasRepl,
-    buildOpt: Option[Build.Successful]
+    successfulBuilds: Seq[Build.Successful]
   ): Either[BuildException, Unit] = either {
 
     val setupPython = options.notForBloopOptions.python.getOrElse(false)
 
     val cache             = options.internal.cache.getOrElse(FileCache())
     val shouldUseAmmonite = options.notForBloopOptions.replOptions.useAmmonite
 
-    val scalaParams = artifacts.scalaOpt match {
-      case Some(artifacts) => artifacts.params
-      case None            => ScalaParameters(Constants.defaultScalaVersion)
+    val scalaParams: ScalaParameters = value {
+      val distinctScalaArtifacts = allArtifacts.flatMap(_.scalaOpt).distinctBy(_.params)
```

Comment:
```suggestion
      val distinctScalaParams = allArtifacts.flatMap(_.scalaOpt).map(_.params).distinct
```
Since you always map to `params` anyway